### DISCUSSION TITLE
Code structure

### DISCUSSION THREAD

**Discussion by Bostwickenator** [lukaszliniewicz/Pandrator#34] [General]
Hey great work on this. The results are quite impressive. I'm finding the code structure pretty rough though. How likely would you be to review and accept an MR that breaks some of this up into separate functional components and reduces some of the duplication?

**Comment by lukaszliniewicz** [comment]
That would be splendid! It really does need some cleaning-up, and I don't have time for it at the moment. I was rushing to include new things I needed (and I'm still doing that, to be honest, trying to get OCR/layout detection to work based on a new model, Surya) - plus I'm incompetent ^^. If you're willing to help, then by all means please do!
